src/Renderer/LayeredMaterial.js

```diff
@@ -241,18 +241,21 @@ LayeredMaterial.prototype.setTexture = function(texture, layer, slot, pitScale)
 
     // TEMP
     if (texture === null) {
-        var l = this.getIdLayer(slot);
-        if (l)
+        var l = this.slotToLayerIndex(slot);
+        if (l >= 0) {
             this.paramLayers[l].z = 0;
+        }
     }
 
 };
 
-LayeredMaterial.prototype.getIdLayer = function(slot) {
-    for (var l = 0; l < this.paramLayers.length; l++) {
-        if (slot === this.paramLayers[l].x)
-            return l;
+LayeredMaterial.prototype.slotToLayerIndex = function(slot) {
+    for (var l = 1; l < this.uniforms.nColorLayer.value; l++) {
+        if (slot < this.paramLayers[l].x) {
+            return l-1;
```

Comment:
Because `this.paramLayers[l].x` is an offset (layer N start at index this.paramLayers[N].x)
So if `slot < this.paramLayers[N].x` that means it belongs to the previous layer, hence the `-1`